.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,39 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "**" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build-jar:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+      - name: Execute Gradle build
+        run: |
+          chmod +x ./gradlew
+          ./gradlew clean build fatJar -x test
+      - uses: actions/upload-artifact@v3
+        with:
+          name: test-telemetry-generator-all.jar
+          path: build/libs/test-telemetry-generator-all.jar
+  docker-image-ci:
+    needs: build-jar
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: test-telemetry-generator-all.jar
+          path: build/libs
+      - name: Build the Docker image
+        run: docker build . --file Dockerfile --tag test-telemetry-generator:$(date +%s)
```

Comment:
will this only check if the build is working or will this also upload the docker image to the registry?

Reply:
Only if the build is working. I have to figure out how to publish the images after this is merged but it will definitely involve some kind of key that we don't want to expose. Will check if GH has additional workflows for pushing the image but that will be a separate PR.

Reply:
There are other projects within cisco-open you can take a look at, e.g.

https://github.com/cisco-open/libnasp/blob/main/.github/workflows/docker-echo-server.yaml

If I remember correctly secrets.GITHUB_TOKEN should all you need to publish to github registry, docker registry is a different discussion